libraries/AP_Scripting/applets/arming-checks.lua

```diff
@@ -0,0 +1,620 @@
+    local foll_ofs_z = Parameter("FOLL_OFS_Z"):get()
+    if foll_ofs_x == 0 and foll_ofs_y == 0 and foll_ofs_z == 0 then
+        return false
+    end
+    return true
+end
+
+-- for many use cases if MNTx_SYSID_DEFLT is set is should match the FOLL_SYSID (remember set ARM_MNTX_SYSID severity to NONE to disable this check)
+local function mntx_sysid_match()
+    local foll_enable = FOLL_ENABLE:get()
+    local mnt1_enable = (Parameter("MNT1_TYPE"):get() ~= 0)
+    local mnt2_enable = (Parameter("MNT2_TYPE"):get() ~= 0)
+
+    if foll_enable ~= 1 then
+        return true
+    end
+    local foll_sysid = FOLL_SYSID:get() or 0
+    if mnt1_enable then
+        local mnt1_sysid = Parameter("MNT1_SYSID_DFLT"):get() or 0
+        if mnt1_sysid > 0 and foll_sysid ~= mnt1_sysid then
+            return false
+        end
+    end
+    if mnt2_enable then
+        local mnt2_sysid = Parameter("MNT2_SYSID_DFLT"):get() or 0
+        if mnt2_sysid > 0 and foll_sysid ~= mnt2_sysid then
+            return false
+        end
+    end
+    return true
+end
+
+-- is the RTL altitude legal? This defaults to 120m (400') as set in ARM_ALT_LEGAL
+local function rtl_altitude_legal()
+    -- plane uses RTL_ALTITUDE in m
+    -- RTL_ALTITUDE will be correct for either plane or copter
+    if (RTL_ALTITUDE ~= nil and (RTL_ALTITUDE:get() or 300) > alt_legal_max) then
+        return false
+    end
+    return true
+end
+
+-- is the Q RTL altitude (quadplane) legal? This defaults to 120m (400') as set in ARM_ALT_LEGAL
+local function q_rtl_alt_legal()
+    -- Plane specific Parameters
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    -- QuadPlane specific Paraemters
+    local q_rtl_alt = Parameter("Q_RTL_ALT"):get() or 0
+    if q_rtl_alt > alt_legal_max then
+        return false
+    end
+    return true
+end
+
+-- is the RTL climb minimum legal? This defaults to 120m (400')
+local function rtl_climb_legal()
+    if (RTL_CLIMB_MIN ~= nil and RTL_CLIMB_MIN:get() > alt_legal_max) then
+        return false
+    end
+    return true
+end
+
+-- Display a message as to whether failsafe will QLand or QRTL
+local function qland_warning()
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    -- Only applies to QuadPlane
+    local q_options = Parameter("Q_OPTIONS"):get() or 0
+    local q_land_option = ((q_options & (1 << 5)) > 0)
+    local q_RTL_option = ((q_options & (1 << 20)) > 0)
+
+    if (not q_land_option) and (not q_RTL_option) then
+        return false
+    end
+    return true -- always ok if not a QuadPlane
+end
+local function qrtl_warning()
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    local q_options = Parameter("Q_OPTIONS"):get() or 0
+    local q_land_option = ((q_options & (1 << 5)) > 0)
+    local q_RTL_option = ((q_options & (1 << 20)) > 0)
+    if q_land_option or q_RTL_option then
+        return false
+    end
+    return true -- always ok if not a QuadPlane
```

Comment:
Comment is out of date, QP early return is now at the top.

Reply:
fixed